### ISSUE TITLE
ui, sql: Logical Plans: discrepancy between sample and explain?

### ISSUE THREAD

**Issue by celiala** [cockroachdb/cockroach#34594]
I sometimes see a logical plan whose last node is "Row source to plan node" with no attributes. But when I run that same query as an `EXPLAIN` in the SQL CLI, I don't see this node, and instead get slightly different nodes in its place. 

Taking this query as an example:
```
select id as movie_id, title, (select name from actors where name = 'Tom Hanks') from movies;
```

In the Admin UI, the sampled plan looks like:
```
plan tree in Admin UI
+---------------------
  root                
   ├── render         
   │    └── scan      
   │                  
   └── subquery       
        └── max1row   
             └── row source to plan node 
```
But when I run `EXPLAIN` for this SQL, the SQL CLI shows this tree:
```
         tree         |    field     |                    description
+---------------------+--------------+----------------------------------------------------+
  root                |              |
   ├── render         |              |
   │    └── scan      |              |
   │                  | table        | movies@primary
   │                  | spans        | ALL
   └── subquery       |              |
        │             | id           | @S1
        │             | original sql | (SELECT name FROM actors WHERE name = 'Tom Hanks')
        │             | exec mode    | one row
        └── max1row   |              |
             └── scan |              |
                      | table        | actors@primary
                      | spans        | ALL
                      | filter       | name = 'Tom Hanks'
```

For Reference, actual snapshot from Admin UI (single paths are shifted left to save on horizontal space):
<img src="https://user-images.githubusercontent.com/3051672/52305912-5c172580-2964-11e9-862f-4c02dd251d8b.png" width="400"/>

**Comment by celiala** [comment]
Discussed with @jordanlewis @knz , and this is by design (see full explanation below).

So next step would be to document this somewhere, as users will likely ask this same question (cc @piyush-singh @Amruta-Ranade ).

Q: Why do we see "row source to planNode", when `EXPLAIN` shows something else?

A: This happens because [when we execute the query, versus simply running an `EXPLAIN`] the DistSQL engine replaces the planNodes that it supports with the DistSQL physical processor equivalents, and while doing so, modifies the underlying plan. 

This only happens when there's a subtree that's only partially supported by DistSQL processor equivalents, since in those cases, the system links the supported DistSQL processors to the unsupported planNodes, and links the end of those unsupported planNodes back up to the supported DistSQL processors that are their input - doing this modifies the unsupported planNodes' sources to point at this "row source to planNode thing", which is a wrapper that converts DistSQL processor to planNode.

**Comment by celiala** [comment]
Oops -- wrote too soon. Jordan, Raphael, and myself are still in discussion as to next steps forward. Will update when we come to consensus.

**Comment by celiala** [comment]
So, there are a few things going on here:

1) We're seeing `row source to planNode` because we don't [yet] have a DistSQL physical processor equivalent for the previous node. [Using above example, we don't have an equivalent for `max1row`, so that's why we see `row source to plan node` as the node directly after/below].

2) In the specific example above, there was additional bug in the explain tree walker: Within a subquery specifically, the tree walker (aka the logic that walks the logical plan/tree to generate explain) would stop when it found a `row source to planNode`, and not continue walking further down the plan tree. 

Issue 2 has since been fixed by #34882.

For Issue 1, the solution is to figure out how best to convey this to the user (eg a UI/documentation solution). @piyush-singh @jordanlewis and @celiala will discuss offline.
